pkg/terraform/executor.go

```diff
@@ -138,6 +138,17 @@ func (ex *Executor) Apply() error {
 	return ex.Execute("apply", "-auto-approve")
 }
 
+// ApplyArgs is a wrapper function that runs
+// `terraform apply -auto-approve args...`.
+func (ex *Executor) ApplyArgs(args ...string) error {
+	ex.logger.Println("Applying part of the Terraform configuration. It might take a long time...")
```

Comment:
I don't like that such basic function does the logging. It may also not be applying part of the Terraform configuration.

I'm not sure if we should add this function, for Terraform executor, having `Apply()`, `Destroy()` etc seems already too much for me.

If we decide to make it this way, maybe we could rename it to `ApplyAutoApprove(args []string)`?

Reply:
Would this solve your concern? Make it `ApplyTargets(args...string)` and it basically does a `terraform apply -auto-approve -target=args[0] -target=args[1] ...`?

I think while very reduced scoped, it seems better. It can't be misused, as it's quite clear when you want to use that instead of other functions, and would do just what we need (no more, no less).

What do you think?

Reply:
It seems to me, `ApplyTargets` is a bit better, as it operates on different level of abstraction (Terraform-scoped, not shell-scoped). And what about logging then?

Reply:
Logging stays the same, ApplyTarget() is applying part of your terraform configuration :)

And both are terraform scoped. **None** should ever be shell-scoped (in the sense of the command being run in a shell)

Reply:
Can we maybe change the logging to mention which targets will be applied, instead of using "part of"?